### ISSUE TITLE
Drop down menu for errors

### ISSUE THREAD

**Issue by noejn2** [Almenon/AREPL-vscode#341]
Sometimes it is annoying to see the errors on the AREPL window. It was especially frustrating when I did not have the option of only updating with shortcut keybinding. I still want to see the code updating on realtime as well as the errors, Would it be possible to have a dropdown menu showing the errors instead of the current setting?

**Comment by Almenon** [comment]
Hi Noé, thanks for the issue report. Yes, that would be possible. I can see the benefit in it too.

What errors are annoying you? Is it the syntax errors / name errors that pop up commonly while typing, or errors in general? If it's just the first I have a option to ignore syntax / name errors.

**Comment by noejn2** [comment]
General errors. For syntax errors that pop up automatically as I type I use the Show Syntax errors option plus oneKeybinding. The latter option prevents me from seeing my coding update automatically though. 

I am thinking about a dropdown option that displays how many errors (and perhaps type of errors) I have in my code, so I see my code stacktrace automatically.

Thanks!

**Comment by Almenon** [comment]
I see. I want to focus on bringing more arepl functionality inline with the code. Would a error message that appears when you hover over the line with a error icon also be a solution? Or something similar to that concept - I would need to play around with the VSCode API to see what I could accomplish.

**Comment by noejn2** [comment]
I disable `arepl.inlineResults` as part of my personalization. 

I exactly do not know what you mean by hovering inline with the code, but I could better explain my issue with the automatic updating:

Error messages are too long and push down information I often refer to. While error information is necessary, I would like to have it as reference rather than popping up every time I do an error or while I am typing. That is why I am suggesting a dropdown menu that updates as I type, but does not push down all I have printed (which I often use as reference) and the variables and data frames.

**Comment by Almenon** [comment]
Ah, okay.

Here's a mockup of the potential UI, what do you think?
Error uncollapsed: https://i.imgur.com/Df4FvSc.png
Error collapsed: https://i.imgur.com/pZmIkus.png
No error: https://i.imgur.com/ZMGo9yH.png

**Comment by noejn2** [comment]
Yes! If it marked the numbers errors, it would improve a lot. So I can keep a look of the errors as they happen to quickly refer to them.

Thanks!

**Comment by Almenon** [comment]
Welcome :)

I don't get why you want the number of errors. Either the code completes or it doesn't. An exception may have multiple causes but in the end there is only one top-level exception.
